### ISSUE TITLE
composer require tries to parse next package as version constraint

### ISSUE THREAD

**Issue by teohhanhui** [composer/composer#6704]
My `composer.json`:

```json
{
    "type": "project",
    "license": "proprietary",
    "require": {
        "php": "^7.1.3",
        "api-platform/api-pack": "^1.0",
        "symfony/flex": "^1.0",
        "symfony/framework-bundle": "^3.3",
        "symfony/yaml": "^3.3"
    },
    "require-dev": {
        "symfony/dotenv": "^3.3"
    },
    "config": {
        "preferred-install": {
            "*": "dist"
        },
        "sort-packages": true
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "App\\Tests\\": "tests/"
        }
    },
    "scripts": {
        "auto-scripts": {
            "make cache-warmup": "script",
            "assets:install %PUBLIC_DIR%": "symfony-cmd"
        },
        "post-install-cmd": [
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
    "conflict": {
        "symfony/symfony": "*",
        "symfony/twig-bundle": "<3.3",
        "symfony/debug": "<3.3"
    },
    "extra": {
        "symfony": {
            "id": "01BS8BV0R9TFCHWS9ASH10T74A",
            "allow-contrib": false
        }
    }
}
```

Output of `composer diagnose`:

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer --no-plugins require ext-apcu ext-ds
```

I get the following output:

```
                                                                              
  [UnexpectedValueException]                                                  
  Could not parse version constraint ext-ds: Invalid version string "ext-ds"  
                                                                              

require [--dev] [--prefer-source] [--prefer-dist] [--no-progress] [--no-suggest] [--no-update] [--no-scripts] [--update-no-dev] [--update-with-dependencies] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--] [<packages>]...
```

And I expected this to happen:

`composer.json` should be:

```json
{
    ...
    "require": {
        ...
        "ext-apcu": "*",
        "ext-ds": "*",
        ...
    },
    ...
}
```

The documentation for the [require](https://github.com/composer/composer/blob/edece864e7e4c668dcad6601df70777882d22116/doc/03-cli.md#require) command does not state that there is such a possibility of parsing the next token as the version constraint...

**Comment by jvasseur** [comment]
Composer does indeed support passing versions as separate arguments event if not documented.

It currently use the presence of a `/` to test if the next argument is a package or a version (https://github.com/composer/composer/blob/master/src/Composer/Package/Version/VersionParser.php#L50) we should check for [PLATFORM_PACKAGE_REGEX](https://github.com/composer/composer/blob/master/src/Composer/Repository/PlatformRepository.php#L27) to make it work with platform packages.

**Comment by alcohol** [comment]
You should always specify the version, even if it is just '*'. If composer cannot find a version as part of the first argument, it will assume the second argument is the version.

```
composer require ext-foo:\* ext-bar:\*
```

**Comment by Seldaek** [comment]
Fixed by 71c2ecbacefd91864b8b0f349b874e37dce00da9
